client/rpc/debug/src/lib.rs

```diff
@@ -57,49 +96,89 @@ where
 	C::Api: DebugRuntimeApi<B>,
 	C::Api: EthereumRuntimeRPCApi<B>,
 {
-	fn trace_transaction(
-		&self,
+	pub fn task(
+		client: Arc<C>,
+		backend: Arc<BE>,
+		frontier_backend: Arc<fc_db::Backend<B>>,
+		permit_pool: Arc<Semaphore>,
+	) -> (impl Future<Output = ()>, DebugRequester) {
+		let (tx, mut rx): (DebugRequester, _) =
+			sp_utils::mpsc::tracing_unbounded("debug-requester");
+
+		let fut = async move {
+			loop {
+				if let Some(((transaction_hash, params), response_tx)) = rx.next().await {
+					let client = client.clone();
+					let backend = backend.clone();
+					let frontier_backend = frontier_backend.clone();
+					let permit_pool = permit_pool.clone();
+					tokio::task::spawn(async move {
```

Comment:
Can you add a comment on the code to specify what mechanism is used to prevent spawning an huge amount of tasks if receiving a huge amount of request instantly please